src/containers/footer-lists/footer-lists.js

```diff
@@ -1,46 +1,54 @@
-import React from 'react';
+import React, { useMemo, useState } from 'react';
 import { useSelector } from 'react-redux';
-
+import { useQuery } from '@apollo/client';
 import { Link } from 'react-router-dom';
 import Typography from '@material-ui/core/Typography';
+
 import { getCategoryURL } from '../../pages/home/categories-list/categories-list';
 import { useStyles } from './footer-lists.styles';
-
 import {
   FOOTER_INFORMATION,
   FOOTER_CONTACTS,
   FOOTER_CATALOGS
 } from '../../translations/footer.translations';
 import routes from '../../const/routes';
+import { getAllCategoriesQuery } from './operations/categories.queries';
+import errorOrLoadingHandler from '../../utils/errorOrLoadingHandler';
 import { countPerPage } from '../../configs';
 
 const { pathToContacts } = routes;
 
 const FooterLists = () => {
   const styles = useStyles();
-  const { categories, language, contacts, quantityPerPage } = useSelector(
-    ({ Categories, Language, Contacts, Products }) => ({
-      categories: Categories.list,
+  const [categories, setCategories] = useState([]);
+  const { language, contacts, quantityPerPage } = useSelector(
+    ({ Language, Contacts, Products }) => ({
       language: Language.language,
       contacts: Contacts.contacts,
       quantityPerPage: Products.countPerPage
     })
   );
 
-  const categoriesList = categories.length
-    ? categories.map(({ _id, name }) => (
-      <div key={_id}>
-        <Typography variant='subtitle2'>
-          <Link
-            className={styles.cardLink}
-            to={`/${getCategoryURL(name)}?page=1&${countPerPage}=${quantityPerPage}`}
-          >
-            {name[language].value}
-          </Link>
-        </Typography>
-      </div>
-    ))
-    : null;
+  const { loading, error } = useQuery(getAllCategoriesQuery, {
+    onCompleted: (data) => setCategories(data.getAllCategories.items)
+  });
+
+  const categoriesList = useMemo(
```

Comment:
seen this in another PR, investigated situation and decided to use it, because here application renders multiple components from array. And if this array will grow, using this hook will help us to optimize application

Reply:
Can you explain me what for is useMemo?

Reply:
It "saves" the result of function, so application doesn't need to recompute it if function is called with the same arguments.
Shoud i remove it?